contentcuration/contentcuration/frontend/channelEdit/components/ContentNodeEditListItem.vue

```diff
@@ -1,80 +1,93 @@
+            draggable,
+            draggableMetadata: contentNode,
+            effectAllowed: draggableEffectAllowed,
+          }"
+          :aria-selected="selected"
+          class="content-node-edit-item"
+          @infoClick="$emit('infoClick', $event)"
+          @topicChevronClick="$emit('topicChevronClick', $event)"
+        >
+          <template #actions-start="{ hover }">
+            <VListTileAction class="handle-col" :aria-hidden="!hover" @click.stop>
+              <transition name="fade">
+                <VBtn :disabled="copying" flat icon>
+                  <Icon color="#686868">
+                    drag_indicator
+                  </Icon>
+                </VBtn>
+              </transition>
+            </VListTileAction>
+            <VListTileAction class="mx-2 select-col" @click.stop>
+              <Checkbox
+                v-model="selected"
+                :disabled="copying"
+                class="mt-0 pt-0"
+                @dblclick.stop
+              />
+            </VListTileAction>
+          </template>
 
-        <template #actions-end>
-          <VListTileAction :aria-hidden="!active" class="action-icon px-1">
-            <Menu v-model="activated">
-              <template #activator="{ on }">
-                <IconButton
-                  icon="optionsVertical"
-                  :text="$tr('optionsTooltip')"
-                  size="small"
-                  :disabled="copying"
-                  v-on="on"
-                  @click.stop
+          <template #actions-end>
+            <VListTileAction :aria-hidden="!active" class="action-icon px-1">
+              <Menu v-model="activated">
+                <template #activator="{ on }">
+                  <IconButton
+                    icon="optionsVertical"
+                    :text="$tr('optionsTooltip')"
+                    size="small"
+                    :disabled="copying"
+                    v-on="on"
+                    @click.stop
+                  />
+                </template>
+                <ContentNodeOptions
+                  v-if="!copying"
+                  :nodeId="nodeId"
+                  @node-flagged="showFlagFeedbackModal = true"
                 />
-              </template>
-              <ContentNodeOptions v-if="!copying" :nodeId="nodeId" />
-            </Menu>
-          </VListTileAction>
-        </template>
+              </Menu>
+            </VListTileAction>
+          </template>
+
+          <template #context-menu="{ showContextMenu, positionX, positionY }">
+            <ContentNodeContextMenu
+              :show="showContextMenu"
+              :positionX="positionX"
+              :positionY="positionY"
+              :nodeId="nodeId"
+            />
+          </template>
+        </ContentNodeListItem>
+      </template>
+    </DraggableItem>
+
+    <FlagFeedbackModal
+      v-if="showFlagFeedbackModal"
+      v-model="showFlagFeedbackModal"
+      :nodeId="nodeId"
+    />
```

Comment:
Instead of embedding the modal in each content node list item, having the component emit an event to a common place in the UI, makes more sense.

Reply:
I don't remember why I did this way :thinking: 

@ozer550 will you be willing to refactor this and take this forward?

Reply:
Sounds good :D 

Btw upon trying to remember, my mind trace says that -- 

Most probably I did this because Vue allows emmiting events from a parent to a direct child only. So I had to catch the event on contentnode list item, I was not able to catch the event anywhere else. 

The workaround for this was to utilise an event bus. So to keep things simple I might have done this.